pkg/operator/secretannotator/aws/reconciler.go

```diff
@@ -178,8 +220,55 @@ func (r *ReconcileCloudCredSecret) updateSecretAnnotations(secret *corev1.Secret
 		secretAnnotations = map[string]string{}
 	}
 
-	secretAnnotations[secretconstants.AnnotationKey] = value
+	secretAnnotations[constants.AnnotationKey] = value
 	secret.SetAnnotations(secretAnnotations)
 
 	return r.Update(context.Background(), secret)
 }
+
+// earlyAnnotationIfModeForced will read the CCO config, and go ahead and annotate the secret if
+// there is a forcedCredentialsMode set. It will return the fetched mode (empty string if no mode set),
+// whether the operator is in manual/disabled mode, and any errors encountered.
+func (r *ReconcileCloudCredSecret) earlyAnnotationIfModeForced(secretKey types.NamespacedName) (modeString string, disabled bool, err error) {
+	var mode operatorv1.CloudCredentialsMode
+	mode, err = utils.GetOperatorMode(r.Client, r.Logger)
+	if err != nil {
+		return
+	}
+
+	disabled, err = utils.IsOperatorDisabled(r.Client, r.Logger)
+	if err != nil {
+		r.Logger.WithError(err).Error("failed to determine if CCO disabled")
+		return
+	}
+
+	secret := &corev1.Secret{}
+	err = r.Get(context.Background(), secretKey, secret)
+	if err != nil {
+		r.Logger.WithError(err).Error("failed to get secret")
+		return
+	}
+
+	// Now annotate the secret depending on the mode defined in the config
+	switch mode {
+	case operatorv1.CloudCredentialsModeManual:
+		// no need to annotate, the user/admin is in control
+	case operatorv1.CloudCredentialsModeMint:
+		err = r.updateSecretAnnotations(secret, constants.MintAnnotation)
+		modeString = constants.MintAnnotation
+	case operatorv1.CloudCredentialsModePassthrough:
+		err = r.updateSecretAnnotations(secret, constants.PassthroughAnnotation)
+		modeString = constants.PassthroughAnnotation
+	case "":
+		// Mode not set, no worries; will have to do dynamic permissions checking
+	default:
+		r.Logger.Warnf("Unexpected mode: %s", mode)
+		err = fmt.Errorf("unexpected mode retrieved: %v", mode)
+	}
+
+	if err != nil {
+		r.Logger.WithError(err).Error("failed to update secret annotations")
```

Comment:
This will log this error entry when there is an unexpected mode as well. It does not appear that that is the intention.